src/gromacs/nblib/gmxsetup.cpp

```diff
@@ -246,14 +240,16 @@ std::unique_ptr<GmxForceCalculator> NbvSetupUtil::setupGmxForceCalculator()
 {
     auto gmxForceCalculator_p = std::make_unique<GmxForceCalculator>(system_, options_);
 
-    gmxForceCalculator_p->nbv_ = setupNbnxmInstance();
+    gmxForceCalculator_p->nbv_ = setupNbnxmInstance(system_.topology(), *options_);
+
+    gmxForceCalculator_p->mdatoms_ = setAtomData(system_);
+    setAtomProperties(gmxForceCalculator_p->nbv_, gmxForceCalculator_p->mdatoms_);
 
     // const PairlistSet& pairlistSet = nbv->pairlistSets().pairlistSet(gmx::InteractionLocality::Local);
     // const gmx::index numPairs = pairlistSet.natpair_ljq_ + pairlistSet.natpair_lj_ + pairlistSet.natpair_q_;
     // gmx_cycles_t cycles = gmx_cycles_read();
 
-    matrix box_;
-    gmx::fillLegacyMatrix(system_.box().matrix(), box_);
+    const matrix& box_ = system_.box().legacyMatrix;
```

Comment:
I would still prefer to have this in the ctor.

Reply:
`box_` renamed to `box` to reflect that it is not an internal state anymore. So it is a temporary variable in that scope.